sdk/core/Azure.Core/src/Response.cs

```diff
@@ -41,6 +41,42 @@ public abstract class Response : IDisposable
         /// </summary>
         public virtual ResponseHeaders Headers => new ResponseHeaders(this);
 
+        private BinaryData? _content;
```

Comment:
Why would we not want to cache the instance? I don't think we want to allocate every time somebody pulls onto the property.

Reply:
We do want to cache it in our concrete implementations, not in the abstraction. The abstraction should allow implementations that do their own memory management, e.g. use pooled memory for buffers.

Reply:
We already do memory management in ResponseBodyPolicy (allocating memory stream and buffering).

Reply:
I don't think having a base overrideable implementation is making it worse.

Reply:
You cannot override fields.

Reply:
You can override the Content property.

Reply:
We could make the field `protected` so subclasses could at least reuse the underlying field (I suspect that our subclasses will want to null the cached value out if the `ContentStream` setter).  Does that seem reasonable or should we just accept that most subclasses that would want to customize the behavior here will end up ignoring the `_content` field and the underlying Response object will be a pointer size larger than it needs to be?

Reply:
>I suspect that our subclasses will want to null the cached value out if the ContentStream setter

Hm, yeah. This kind of makes the base caching implementation incorrect in the general case. Because it still requires every implementor to opt into invalidating the cache.

Taking that into account I think we should return a new instance of the BinaryData in the base class and not cache.

Reply:
> Taking that into account I think we should return a new instance of the BinaryData in the base class and not cache. That would at least make the implementation correct and we can make it efficient in the individual implementations.

SGTM!